src/pages/home/report/ReportTypingIndicator.js

```diff
@@ -26,66 +26,54 @@ const defaultProps = {
+        return null;
     }
 
-    render() {
-        const numUsersTyping = _.size(this.state.usersTyping);
+    const numUsersTyping = _.size(usersTyping);
 
-        // If we are offline, the user typing statuses are not up-to-date so do not show them
-        if (this.props.network.isOffline) {
+    // Decide on the Text element that will hold the display based on the number of users that are typing.
+    switch (numUsersTyping) {
+        case 0:
             return null;
-        }
-
-        // Decide on the Text element that will hold the display based on the number of users that are typing.
-        switch (numUsersTyping) {
-            case 0:
-                return null;
-
-            case 1:
-                return (
-                    <TextWithEllipsis
-                        leadingText={PersonalDetails.getDisplayName(this.state.usersTyping[0])}
-                        trailingText={` ${this.props.translate('reportTypingIndicator.isTyping')}`}
-                        textStyle={[styles.chatItemComposeSecondaryRowSubText]}
-                        wrapperStyle={[styles.chatItemComposeSecondaryRow, styles.flex1]}
-                        leadingTextParentStyle={styles.chatItemComposeSecondaryRowOffset}
-                    />
-                );
 
-            default:
-                return (
-                    <Text
-                        style={[styles.chatItemComposeSecondaryRowSubText, styles.chatItemComposeSecondaryRowOffset]}
-                        numberOfLines={1}
-                    >
-                        {this.props.translate('reportTypingIndicator.multipleUsers')}
-                        {` ${this.props.translate('reportTypingIndicator.areTyping')}`}
-                    </Text>
-                );
-        }
+        case 1:
+            return (
+                <TextWithEllipsis
+                    leadingText={PersonalDetails.getDisplayName(usersTyping[0])}
+                    trailingText={` ${props.translate('reportTypingIndicator.isTyping')}`}
+                    textStyle={[styles.chatItemComposeSecondaryRowSubText]}
+                    wrapperStyle={[styles.chatItemComposeSecondaryRow, styles.flex1]}
+                    leadingTextParentStyle={styles.chatItemComposeSecondaryRowOffset}
+                />
+            );
+
+        default:
+            return (
+                <Text
+                    style={[styles.chatItemComposeSecondaryRowSubText, styles.chatItemComposeSecondaryRowOffset]}
+                    numberOfLines={1}
+                >
+                    {props.translate('reportTypingIndicator.multipleUsers')}
+                    {` ${props.translate('reportTypingIndicator.areTyping')}`}
+                </Text>
+            );
     }
 }
 
 ReportTypingIndicator.propTypes = propTypes;
 ReportTypingIndicator.defaultProps = defaultProps;
+ReportTypingIndicator.displayName = 'ReportTypingIndicator';
 
 export default compose(
     withLocalize,
```

Comment:
are we not migrating the withLocalize to a custom hooK with context?

Reply:
@hayata-suenaga I don't know about the use of localized context and am not able to find a reference. Can you please give me any component that we can take reference from?

Reply:
actually I have to check internally about this but for this PR, using HOC for now is good 👍